README.md

```diff
@@ -3,156 +3,351 @@
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 
 ## Introduction
 
-Add an DOM event that will inform applications about storage pressure (low disk space availability).
 
+This proposal aims to provide an early warning when the storage device backing
+the user's data is running low on free space. This warning is provided to
+currently running applications, which can pause or abandon in-progress
+operations that would otherwise fill up the user's storage device.
 
-## Background
 
-Quota is the system which sets and enforces limits on disk usage by the browser and web applications. Up until recently (M80), the quota system would factor in free disk space when determining how much quota each origin is allowed. This allowed applications to rely on the quota system to know exactly how much disk space they had to write to. **Without information about what will and will not fit on disk, applications that depend on storage usage cannot guarantee a working experience to their users.**
+## Background
 
-In the current design, free disk is no longer incorporated into the calculation of an application’s storage quota due to a security issue. This leaves applications and developers in a place where they have no guarantees around how much space they can write to, and must wait for write errors before realizing they need to scale back their offline caching.
+The Storage standard
+[recommends](https://storage.spec.whatwg.org/#storage-quota) that
+"[the storage quota of a storage shelf] should be less than the total
+available storage space on the device to give users some wiggle room." In the
+past, Chrome treated this recommendation as a requirement, and used the free
+disk space as a hard cap for quota. In other words, for certain users, a
+site's quota would match the available disk space.
+
+Chrome's approach became [a security problem](https://crbug.com/960305) when
+combined with the
+[API exposed in the Storage Standard](https://storage.spec.whatwg.org/#api),
+which includes
+[a quota estimate method](https://storage.spec.whatwg.org/#dom-storagemanager-estimate).
+The method allows sites to obtain their available quota which, for some Chrome
+users, used to reflect their free disk space. The ability to precisely
+measure free disk space is a security problem, because it can be used to measure
+the size of cross-origin resources. The same ability may also be used for
+fingerprinting.
+
+In order to fix the security problem described above, Chrome switched to
+granting sites a fixed amount of quota, which does not account for the amount of
+free disk space.
+
+While this change fixed the security problem, it was a regression for web
+applications that relied on the details of Chrome's implementation of the
+Storage API to provide useful functionality to their users.
+
+One class of regressions is avoidig downloading a large corpus of data when
+the free disk space is low. For example, some Google properties would not
```

Comment:
Should we just say "some applications" instead of "some Google properties"?

Reply:
Good point!

I was trying to hint that we have specific use cases. After a second read, I agree that the phrasing above fails to meet that goal, and instead gives off the impression that we're only considering Google use cases. I think (and hope) that the _Motivating use cases_ section has enough specific examples.

Reply:
It seems like the phrasing is unchanged, but I could be missing something. Could you explain what the change was?

Reply:
The old phrasing was _"One class of regressions is avoidig downloading a large corpus of data when
the free disk space is low."_ The new phrasing is _"One class of user experience regressions is that applications are no longer able to avoid downloading a large corpus of data when the free disk space is low."_

The diff around these comments is marked as "Outdated." You may need to refresh your page / go to a different view. Sorry, I don't know a lot about what could go wrong with GitHub :smile: